azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/configurations/DeploymentType.java

```diff
@@ -29,17 +29,17 @@
 
     public static DeploymentType fromString(final String input) {
         if (StringUtils.isEmpty(input)) {
-            return MS_DEPLOY;
+            return ZIP;
         }
 
         switch (input.toLowerCase(Locale.ENGLISH)) {
             case ConstantValues.FTP_VALUE:
                 return FTP;
-            case ConstantValues.ZIP_VALUE:
-                return ZIP;
             case ConstantValues.MS_DEPLOY_VALUE:
-            default:
                 return MS_DEPLOY;
+            case ConstantValues.ZIP_VALUE:
+            default:
```

Comment:
The same concern applies to the default branch. We should throw an error if we cannot process an input.

Reply:
Agree, just use the default value to proceed the deployment is not a good practice for users to know what is wrong. Issue #308 was created to track it since web app and function app have the same behavior and we need to fix them both.